frontend/src/layout/navigation-3000/sidepanel/panels/access_control/AccessControlObject.tsx

```diff
@@ -0,0 +1,440 @@
+import { IconX } from '@posthog/icons'
+import {
+    LemonBanner,
+    LemonButton,
+    LemonDialog,
+    LemonInputSelect,
+    LemonSelect,
+    LemonSelectProps,
+    LemonTable,
+} from '@posthog/lemon-ui'
+import { BindLogic, useActions, useAsyncActions, useValues } from 'kea'
+import { PayGateMini } from 'lib/components/PayGateMini/PayGateMini'
+import { upgradeModalLogic } from 'lib/components/UpgradeModal/upgradeModalLogic'
+import { UserSelectItem } from 'lib/components/UserSelectItem'
+import { LemonTableColumns } from 'lib/lemon-ui/LemonTable'
+import { LemonTableLink } from 'lib/lemon-ui/LemonTable/LemonTableLink'
+import { ProfileBubbles, ProfilePicture } from 'lib/lemon-ui/ProfilePicture'
+import { capitalizeFirstLetter } from 'lib/utils'
+import { useEffect, useState } from 'react'
+import { urls } from 'scenes/urls'
+import { userLogic } from 'scenes/userLogic'
+
+import {
+    AccessControlType,
+    AccessControlTypeMember,
+    AccessControlTypeRole,
+    AvailableFeature,
+    OrganizationMemberType,
+} from '~/types'
+
+import { accessControlLogic, AccessControlLogicProps } from './accessControlLogic'
+
+export function AccessControlObject(props: AccessControlLogicProps): JSX.Element | null {
+    const { canEditAccessControls, humanReadableResource } = useValues(accessControlLogic(props))
+
+    const suffix = `this ${humanReadableResource}`
+
+    return (
+        <BindLogic logic={accessControlLogic} props={props}>
+            <div className="space-y-4">
+                {canEditAccessControls === false ? (
+                    <LemonBanner type="info">
+                        <b>You don't have permission to edit access controls for {suffix}.</b>
+                        <br />
+                        You must be the creator of it, a Project Admin, or an Organization Admin.
+                    </LemonBanner>
+                ) : null}
+                <h3>Default access to {suffix}</h3>
+                <AccessControlObjectDefaults />
+
+                <h3>Members</h3>
+                <PayGateMini feature={AvailableFeature.PROJECT_BASED_PERMISSIONING}>
```

Comment:
It's a pretty nasty loading state here... There is a good few seconds delay due to billing api being so slow and you just see a blank... Could be follow up but I think `PayGateMini` needs a "loading" prop so you can render some skeletons

Reply:
Yes this is really needed. Thanks for the ping.